File: ABC/ABC154/abc154_d.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(int i = 1; i <= n; i++)
using namespace std;

signed main(void)
{
	int n, k;
	double sum[200010];
	double ans;

	cin >> n >> k;
	ans = 0;
	rep(i, n)
	{
		int a;
		cin >> a;
		sum[i] = (double)(a + 1) / 2;
	}
	rep(i, n) sum[i] += sum[i - 1];
	for (int i = k; i <= n; i++)
		ans = max(sum[i] - sum[i - k],ans);
	cout << fixed << setprecision(12) << ans << endl;
	return 0;
}

File: ABC/ABC154/abc154_d.py
N, K = map(int, input().split())
p = list(map(int, input().split()))

ps = [(1 + p[i]) / 2 for i in range(N)]
sums = [0] * (N + 1)
for i in range(N):
    sums[i+1] = sums[i] + ps[i]
ans = 0
for i in range(N-K+1): 
    ans = max(ans, sums[i+K] - sums[i])
print(ans)
